### ISSUE TITLE
Turing the `b3sum` crate into a library (with an executable)

### ISSUE THREAD

**Issue by bhgomes** [BLAKE3-team/BLAKE3#132]
I'm planning on using BLAKE3 in a project but really only want to use the `b3sum` functionality. Would it be possible to turn the `b3sum` crate into a library (with an executable) so that I can use say the `write_hex_output` function instead of having a copy of it in my library? Right now, I am relying on the `b3sum` executable to be installed and running a new process every time I want to use it.

**Comment by oconnor663** [comment]
I'm not necessarily opposed, though if it was just for `write_hex_output` I'm not sure it would be worth it. I think the real meat of `b3sum` is in the memory mapping and multithreading that it does. It's not nearly as complicated as what's in `blake3` itself, but it's still a little much for folks who haven't dealt with memory mapping before, especially because it necessarily involves unsafe code. Exposing a function like `b3sum::hash_this_file_the_fastest_way_you_can_including_maybe_multithreading()` might be a win.

**Comment by bhgomes** [comment]
Right, I meant that function as a simple use case for my application, but whatever makes sense for the user base is great.

**Comment by oconnor663** [comment]
Well so far the user base is...pretty much you :) Or at least, you're the only one who's asked us about it so far. Do you think you'd have time to put together a PR for this? I'd be happy to review it and help you get it landed.

**Comment by bhgomes** [comment]
Yeah, I'll try to put together the first draft PR by this weekend. Thanks.
